pgmpy/sampling/base.py

```diff
@@ -418,3 +419,35 @@ def _return_samples(return_type, samples):
             return samples
     else:
         return samples
+
+
+def _map_to_state_name(model, samples):
+    """
+        A utility function to map samples to corresponding state name
+    """
+    # Assuming all cpd in model either have state name or not
+
+    method = 'get_cpds' if isinstance(model, BayesianModel) else 'get_factors'
+
+    if getattr(model, method)(samples.dtype.names[0]).state_names is not None:
+
+        if '_weight' in samples.dtype.names:
+            index = len(samples.dtype.names) - 1
+            types = [(var_name, object) for var_name in samples.dtype.names[:-1]]
+            types.append(('_weight', 'float'))
+        else:
+            types = [(var_name, object) for var_name in samples.dtype.names]
+            index = len(samples.dtype)
+
+        named_samples = np.zeros(samples.size, dtype=types).view(np.recarray)
+
+        for node in samples.dtype.names[:index]:
+            cpd = getattr(model, method)(node)
+            named_samples[node] = np.array(list(map(lambda x: cpd.state_names[node][x], samples[node])))
```

Comment:
@khalibartan Can use `np.fromiter`. That should be faster.

Reply:
Or a faster approach would be to create an array with state names and then just index over it. For example:
```python
states = np.array(['state_1', 'state_2'])
samples = np.array([0, 0, 1, 1, 0, 1, 0])
state_samples = states[samples]
```

Reply:
@ankurankan I tried using np.fromiter but it gave the error regarding variable size while assigning to the array. Lemme check for the second suggestion though.